src/Export/CacheHandler.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FINDOLOGIC\FinSearch\Export;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+
+class CacheHandler
+{
+    private const CACHE_ID_PRODUCT_GROUP = 'fl_product_groups';
+    private const CACHE_LIFETIME_PRODUCT_GROUP = 60 * 11;
+
+    /** @var CacheItemPoolInterface */
+    protected $cache;
+
+    /** @var string */
+    protected $shopkey;
+
+    public function __construct(CacheItemPoolInterface $cache)
+    {
+        $this->cache = $cache;
+    }
+
+    public function setShopkey(string $shopkey): void
+    {
+        $this->shopkey = $shopkey;
+    }
+
+    /**
+     * If we have reached the last page of the dynamic product group export, we set a flag in cache to
+     * know that the dynamic product groups are warmed up.
+     */
+    public function dynamicProductGroupWarmUp(int $start, int $count, int $total = 0): void
```

Comment:
A method name should always start with a verb.

```suggestion
    public function warmUpDynamicProductGroups(int $start, int $count, int $total = 0): void
```